docs/rfcs/1020-pki-support.md

```diff
@@ -0,0 +1,291 @@
+[`pki_enrollment_load_peer_certificate`]: https://github.com/Scille/parsec-extensions/blob/5f0ae5ac44d23d775d92e241b546fc0cac327a0a/parsec_ext/smartcard/__init__.py#L336-L355
+[`pki_enrollment_load_submit_payload`]: https://github.com/Scille/parsec-extensions/blob/5f0ae5ac44d23d775d92e241b546fc0cac327a0a/parsec_ext/smartcard/__init__.py#L385-L406
+[`pki_enrollment_load_accept_payload`]: https://github.com/Scille/parsec-extensions/blob/5f0ae5ac44d23d775d92e241b546fc0cac327a0a/parsec_ext/smartcard/__init__.py#L409-L430
+[`pki_enrollment_load_local_pending_secret_part`]: https://github.com/Scille/parsec-extensions/blob/5f0ae5ac44d23d775d92e241b546fc0cac327a0a/parsec_ext/smartcard/__init__.py#L433-L473
+
+## Design
+
+Here is a possible user flow of Alice joining the organization CoolOrg using the PKI to enable the async enrollment feature:
+
+ 1. Alice wants to join CoolOrg organization.
+ 2. Alice already has a PKI configured on its system and the link to submit the join request.
+ 3. Alice select the certificate to use for the enrollment ([`pki_enrollment_select_certificate`])
+
+    > [!IMPORTANT]
+    > The certificate need to have a `CommonName` ([oid-2.5.4.3](https://oid-base.com/get/2.5.4.3)) and a `EmailAddress` ([oid-1.2.840.113549.1.9.1](https://oid-base.com/get/1.2.840.113549.1.9.1)) attributes in the certificate's subject.
+
+ 4. Alice creates an encryption (for the user certificate) and signing key pair (for the device certificate).
+ 4. Alice save the private parts of the key pairs on the filesystem ([`pki_enrollment_create_local_pending`]).
+ 5. Alice submits the request (authorized to re-submit later on with updated information) with: ([`pki_enrollment_submit`]):
+
+    - The PKI certificate.
+    - The signed payload consisting of the public key pairs parts and the desired device label ([`PkiEnrollmentSubmitPayload`]) (signed using [`pki_enrollment_sign_payload`]).
+
+ 6. Mallory (a CoolOrg admin) sees the enrollment request from Alice ([`pki_enrollment_list`]).
+ 7. Mallory verifies the provided PKI certificate ([`pki_enrollment_load_peer_certificate`]) and provided data ([`pki_enrollment_load_submit_payload`]).
+ 8. If Mallory accepts the request (after being validated), it creates the Alice's user and device certificates with the provided verification and encryption keys.
+ 9. Mallory sign both certificate with its device signing key. Then prepare the "accept" response payload ([`PkiEnrollmentAnswerPayload`])
+10. Mallory select a certificate ([`pki_enrollment_select_certificate`])
+11. Mallory sign the response payload with the certificate ([`pki_enrollment_sign_payload`])
+12. Mallory send the signed payload alongside the used certificate to the server ([`pki_enrollment_accept`])
+13. Alice check the server to see the state of the request ([`pki_enrollment_info`])
+14. Alice retrieve the accept response from the server
```

Comment:
Even if not so interesting, maybe we should mention what happen (or how Parsec should behave if rejected).

Reply:
done